File: apps/backend-site/app/models/post.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, DateTime, ForeignKey, Table
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.db.base import Base

# associação N:M Post ↔ Category
post_categories = Table(
    "post_categories",
    Base.metadata,
    Column("post_id",     UUID(as_uuid=True), ForeignKey("posts.id"), primary_key=True),
    Column("category_id", UUID(as_uuid=True), ForeignKey("categories.id"), primary_key=True),
)

class Post(Base):
    __tablename__ = "posts"
    id          = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title       = Column(String, nullable=False, index=True)
    slug        = Column(String, nullable=False, unique=True, index=True)
    author_id   = Column(UUID(as_uuid=True), ForeignKey("authors.id"), nullable=False)
    created_at  = Column(DateTime, default=datetime.utcnow)
    updated_at  = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    thumbnail_url = Column(String, nullable=True)

    author     = relationship("Author", backref="posts")
    categories = relationship("Category", secondary=post_categories, backref="posts")
    blocks     = relationship("Block", order_by="Block.position", backref="post", cascade="all, delete-orphan")

File: apps/clubi.ly/src/types/post.ts
import type { Author } from './author'
import type { Category } from './categoryBlog'

export type BlockType = 'text' | 'image' | 'video' // adicione outros se tiver

export interface Block {
  id: string
  position: number
  type: BlockType
  content: string
}

export interface BlockCreate {
  position: number
  type: BlockType
  content: string
}

export interface Post {
  id: string
  title: string
  slug: string
  author: Author
  categories: Category[]
  blocks: Block[]
  created_at: string
  updated_at: string
  thumbnail_url?: string
}

export interface PostPage {
  items: Post[];
  page: number;
  page_size: number;
  total: number;
}
